mPowerSDK/Dashboard/APHCorrelationsSelectorViewController.m

```diff
@@ -0,0 +1,203 @@
+    }
+    return self;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+}
+
+- (void)viewWillAppear:(BOOL)__unused animated
+{
+    [super viewWillAppear:animated];
+    
+    [self.tableView reloadData];
+    [self moveUpTableView];
+}
+
+- (void)viewWillDisappear:(BOOL)animated
+{
+    [super viewWillDisappear:animated];
+}
+
+#pragma mark - Helper Functions
+
+- (void)dismissView
+{
+    [self moveDownTableView];
+}
+
+- (void)initTableView
+{
+    self.downFrame = CGRectMake(0, self.view.frame.size.height, self.view.frame.size.width, self.view.frame.size.height - 20);
+    self.upFrame = CGRectMake(0, 20, self.view.frame.size.width, self.view.frame.size.height - 20);
+    
+    self.tableView = [[UITableView alloc] initWithFrame:self.downFrame style:UITableViewStylePlain];
+    self.tableView.tableFooterView = [[UIView alloc] initWithFrame:CGRectZero];
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    [self.view addSubview:self.tableView];
+}
+
+- (void)moveUpTableView
+{
+    [UIView
+     animateWithDuration:0.5
+     animations:^{
+         self.tableView.frame = self.upFrame;
+     }];
+}
+
+- (void)moveDownTableView
+{
+    [UIView
+     animateWithDuration:0.5
+     animations:^{
+         self.tableView.frame = self.downFrame;
+     }
+     completion:^(BOOL finished) {
+         [self dismissViewControllerAnimated:NO completion:nil];
+    }];
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)__unused tableView
+{
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)__unused tableView numberOfRowsInSection:(NSInteger)__unused section
+{
+    return self.scoringObjects.count;
+}
+
+#pragma mark - UITableViewDelegate
+
+- (UIView *)tableView:(UITableView *)__unused tableView viewForHeaderInSection:(NSInteger)section
+{
+    NSString *cancelString = NSLocalizedStringWithDefaultValue(@"Cancel", @"APCAppCore", APHLocaleBundle(), @"Cancel", nil);
+    
+    UIButton *cancel = [UIButton buttonWithType:UIButtonTypeCustom];
+    cancel.frame = CGRectMake(self.tableView.frame.size.width - 100, 0, 100, 44.0);
+    [cancel setTitle:cancelString forState:UIControlStateNormal];
+    [cancel setTitleColor:[UIColor appTertiaryRedColor] forState:UIControlStateNormal];
+    [cancel addTarget:self action:@selector(dismissView) forControlEvents:UIControlEventTouchUpInside];
+    cancel.titleLabel.font = [UIFont fontWithName:@"HelveticaNeue" size:17];
+    
+    UITableViewHeaderFooterView *headerView = [[UITableViewHeaderFooterView alloc] init];
+    headerView.textLabel.text = NSLocalizedStringWithDefaultValue(@"Select Category", @"APCAppCore", APHLocaleBundle(), @"Select Category", nil);
+    [headerView addSubview:cancel];
+    
+    return headerView;
+}
+
+- (CGFloat)tableView:(UITableView *)__unused tableView heightForHeaderInSection:(NSInteger)__unused section
+{
+    return 44.0;
```

Comment:
Use `tableView.sectionHeaderHeight`. It's more performant that the hardcoded magic number.